command/node_intro_create.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/posener/complete"
+)
+
+type NodeIntroCreateCommand struct {
+	Meta
+
+	// Fields for the command flags.
+	json     bool
+	tmpl     string
+	ttl      string
+	nodeName string
+	nodePool string
+}
+
+func (n *NodeIntroCreateCommand) Help() string {
+	helpText := `
+Usage: nomad node intro create [options]
+
+  Generates a new node introduction token. This token is used to authenticate
+  a new Nomad client node to the cluster.
+
+  If ACLs are enabled, this command requires a token with the 'node:write'
+  capability.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault|usageOptsNoNamespace) + `
+
+Create Options:
+
+  -node-name
+    The name of the node to which the introduction token will be scoped. If not
+    specified, the value will be left empty.
+
+  -node-pool
+    The node pool to which the introduction token will be scoped. If not
+    specified, the value "default" will be used.
+
+  -ttl
+    The TTL to apply to the introduction token. If not specified, the server
+    configured default value will be used.
+
+  -json
+    Output the response object in JSON format.
+
+  -t
+    Format and display the response object using a Go template.
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (n *NodeIntroCreateCommand) Synopsis() string {
+	return "Generate a new node introduction token"
+}
+
+func (n *NodeIntroCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(n.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
```

Comment:
We have a completion function for node pools, so we can `-node-pool` to this list.

Reply:
Good shout; added within: https://github.com/hashicorp/nomad/pull/26332/commits/19b836ec31c9b2d26c35b211f383e54a45cd04f8